Code/Framework/AzCore/AzCore/Serialization/AZStdContainers.inl

```diff
@@ -1808,7 +1808,13 @@ namespace AZ
                     return false;
                 };
                 EnumElements(instance, captureValue);
-                typename T::value_type *valuePtr = *reinterpret_cast<typename T::value_type**>(ptrToRawPtr);
+
+                if (!ptrToRawPtr)
+                {
+                    return nullptr;
+                }
```

Comment:
Yeah this fix isn't correct.
The `instance` variable should never be nullptr.

Every function that is implemented as part of the derived IDataContainer class that accepts an `instance` expects it to be a non-nullptr.
If it is nullptr, then the issue is in what calls this function.

Reply:
> Yeah this fix isn't correct. The `instance` variable should never be nullptr.

Is this specified anywhere?
I understand if this is not checked since this is a fairly low-level function, but I'm thinking the comment should specify that if it's the case.

I can try to add a check higher up instead.

Reply:
Ok actually, let me specify... the instance variable is not nullptr when the function is called

![image](https://user-images.githubusercontent.com/82231674/164575643-843dcd09-3464-4403-a540-e2e69f7ee02f.png)

The problem is that the object has uninitialized optional parameters, so EnumElements never sets a value to ptrToRawPtr.
Is this a valid case? Cause it sounds like it should be.

Reply:
Well `GetElementByIndex` shouldn't be called with an Optional.
Notice that `bool CanAccessElementsByIndex() const override { return false; }` returns false,

Reply:
Ok let me try that.
I will also try to add some comments to this function to clarify why this was incorrect so that hopefully it doesn't happen again :)

Reply:
Done! Moved the check up one level, results in the same UI and behavior.

Since the change is now part of the RPE I added some sig-content reviewers.